src/data_table/local_data_table.js

```diff
@@ -459,6 +461,56 @@ class LocalDataTable extends View {
     this.dataTable.on("page", this._bulkCheckboxAdjust);
   }
 
+  _setupPaginationHistory() {
+    this.dataTable.on("page", function() {
+      var page = this.dataTable.fnPagingInfo().iPage;
+      if (page !== this._parseQueryString(window.location.search)) {
+        history.pushState({}, "pagination", this._createSearchString(this.dataTable.fnPagingInfo().iPage));
+      }
+    }.bind(this));
+    window.onpopstate = () => {
+      this._pageToSearchPage();
+    };
+    this.on("render", this._afterRender());
+  }
+
+  _afterRender() {
+    this._pageToSearchPage();
+  }
+
+  _pageToSearchPage() {
+    let pageNumber = this._parseQueryString(window.location.search);
+    if (pageNumber >= 0) {
+      this.page(this._parseQueryString(window.location.search));
+    }
+  }
+
+  _createSearchString(pageNumber) {
+    var queryString = window.location.search;
+    var searchString = "?";
+    pageNumber++;
+    var pageString = "page=" + pageNumber;
+    if (queryString[0] === '?') {
+      queryString = queryString.substr(1);
+      var parameters = $.deparam(queryString);
+      parameters.page = pageNumber;
+      searchString += $.param(parameters);
+    } else {
+      searchString += pageString;
+    }
+    return searchString;
+  }
+
+  _parseQueryString(search) {
+    if (search[0] === '?') {
+      search = search.substr(1);
+    }
+    let parameters = $.deparam(search);
```

Comment:
looks like you're toggling back and forth between es5 and es6 in here, solidify in one please.

Reply:
this and the lines above it can be simplified to `let parameters = $.deparam(search.split('?')[1] || '')`